core/gather/session.js

```diff
@@ -44,6 +44,23 @@ class ProtocolSession extends CrdpEventEmitter {
     this._handleProtocolEvent = this._handleProtocolEvent.bind(this);
     // @ts-expect-error Puppeteer expects the handler params to be type `unknown`
     this._cdpSession.on('*', this._handleProtocolEvent);
+
+    // Poor man's Promise.withResolvers()
+    /** @param {Error} _ */
+    let rej = _ => {};
+    const promise = /** @type {Promise<never>} */ (new Promise((_, theRej) => rej = theRej));
+    this._targetCrashedResolver = {promise, rej};
```

Comment:
Could we get away with just storing the `promise` and using `rej` directly on L62?